src/main/auth/service.ts

```diff
@@ -1,19 +1,52 @@
+import { safeStorage } from 'electron';
+import Logger from 'electron-log';
 import ConfigStore, { defaults, fieldsToSave } from '../config';
 import packageConfig from '../../../package.json';
 import { User } from '../types';
 
+const TOKEN_ENCODING = 'latin1';
+
+export function encryptToken() {
+  const bearerTokenEncrypted = ConfigStore.get('bearerTokenEncrypted');
+
+  if (bearerTokenEncrypted) return;
+
+  Logger.info('TOKEN WAS NOT ENCRYPTED, ENCRYPTING...');
+
+  if (!safeStorage.isEncryptionAvailable()) {
+    throw new Error('Safe Storage is not available');
+  }
+
+  const plainToken = ConfigStore.get('bearerToken');
+
+  const buffer = safeStorage.encryptString(plainToken);
+  const encryptedToken = buffer.toString(TOKEN_ENCODING);
+
+  ConfigStore.set('bearerToken', encryptedToken);
+  ConfigStore.set('bearerTokenEncrypted', true);
+}
+
 export function setCredentials(
   userData: User,
   mnemonic: string,
   bearerToken: string
 ) {
+  if (!safeStorage.isEncryptionAvailable()) {
```

Comment:
Could we try to test where (SOs) this feature is not available? Maybe we are forcing some users to change the SO, which could be fatal

Reply:
There are two cases where this could be false:
- We are not on the main process: `safeStorage` is only available from the main process.
- Depending on the OS: only on MacOS can return false if the system does not have access to Keychain. On Windows and Linux once the app is ready is available.


The first one is manageable since we can use the event bus to communicate with the main process. If it fails with the second it should be fixable by the user, I'm not quite sure how the Keychain functions but the user should be able to allow the app to use it